zebedee-reader/src/main/java/com/github/onsdigital/zebedee/reader/api/ReadRequestHandler.java

```diff
@@ -173,17 +134,63 @@ public Resource findResource(HttpServletRequest request) throws ZebedeeException
     public long getContentLength(HttpServletRequest request) throws ZebedeeException, IOException {
         String uri = extractUri(request);
         String collectionId = getCollectionId(request);
+
+        return get(collectionId, uri, 
+                r -> r.getCollectionContentLength(collectionId, RequestUtils.getSessionId(request), uri),
+                r -> r.getPublishedContentLength(uri));
+    }
+
+    /**
+     * Get a resource in either:
+     * <ol>
+     * <li>the given collection (if any) in the current language</li>
+     * <li>published content in the current language</li>
+     * <li>the given collection (if any) in English</li>
```

Comment:
I had a play with it and I think the comment in line 148 is not actually true. It seems to be that if there is English content in both master and in collections, and no Welsh content in either, then if someone tries to create a new Welsh page, it will always use the English content from master (the published content) rather than any that is currently in progress. This seems correct but this means that the comment is not quite true.

Reply:
Thanks for being thorough Ella! 🙂 

It took me a while to figure out why it was behaving as you say when the code seems to be doing what the comment says...

The answer is that the `publishedSupplier` is of type `FileSystemContentReader` whereas the `collectionSupplier` is a `CollectionContentReader`. 
So the `publishedSupplier` will actually try and find the Welsh content and, if it is not there, [will also find the English content](https://github.com/ONSdigital/zebedee/blob/24b122e006a11ebbab38e8377ad9cae95faa628c/zebedee-reader/src/main/java/com/github/onsdigital/zebedee/reader/FileSystemContentReader.java#L399-L401)! Therefore, as far as this handler is concerned, the content was found in published (master) and it never goes into the catch in line 179 which would trigger the bottom 2 scenarios in the list of this comment. It is all really convoluted!!

The scenario in the catch clause is there for when we are creating content for the first time. In that case, we won't have anything in master so neither  `collectionSupplier` nor `publishedSupplier` will find anything, yet we have the English content in the collection (since there is no way of creating Welsh content without the English). Hope this makes sense?